### DISCUSSION TITLE
how to check that wallbox-mqtt is well installed.

### DISCUSSION THREAD

**Discussion by niouniou49** [jagheterfredrik/wallbox-mqtt-bridge#34] [Q&A] [answered]
hi guys, i just installed the wallbox-mqtt-bridge. But it seems mqtt is not able to receive anything.
where can i check logs ?
should i restart power the wallbox?
how can i be sure it works properly?
best pascal

**Comment by jagheterfredrik** [comment]
Do one at a time until you understand what the issue is:

1. Double check your bridge.ini configuration values.
2. Check that your MQTT broker is installed correctly. You can do this using e.g. MQTT Explorer.
3. ssh to the wallbox and run
```
cd mqtt-bridge
systemctl stop mqtt-bridge
venv/bin/python bridge.py
```
Observe the output, stop using ctrl+c and then start the bridge again using `systemctl start mqtt-bridge`.

**Reply by niouniou49** [reply]
thanks a lot
seems the bridge.ini was corrupted with ^M at the end of each lines.
the command 'venv/bin/python bridge.py' doesn't work. not such file or directory
i will do 'systemctl start mqtt-bridge' and see

**Reply by jagheterfredrik** [reply]
Did you do `cd ~/mqtt-bridge`?

**Reply by niouniou49** [reply]
i did from the mqtt-bridge folder. and the result is 
root@WB628598:~/mqtt-bridge# venv/bin/python bridge.py
-sh: venv/bin/python: No such file or directory

**Reply by jagheterfredrik** [reply]
then `./install.sh` did not run 🤷‍♂️

**Reply by jagheterfredrik** [reply]
Oh, I see, what model and firmware are you running?

**Reply by niouniou49** [reply]
pulsar plus version 5.17.87

**Reply by jagheterfredrik** [reply]
Can you SSH to it
```
cd mqtt-bridge
chmod +x install.sh
./install.sh
```
and give me the output

**Reply by niouniou49** [reply]
root@WB628598:~/mqtt-bridge# ./install.sh
Setting up virtual environment
./install.sh: line 24: python3: command not found
Installing Python dependencies
./install.sh: line 27: venv/bin/pip: No such file or directory
Setting up the MQTT bridge systemd service
ln: failed to create symbolic link '/lib/systemd/system/mqtt-bridge.service': File exists
Enable the service to start on boot..
..and launch the service now
Sleeping for 3 seconds before checking the status of the systemd service
● mqtt-bridge.service - MQTT Bridge
     Loaded: loaded (/etc/systemd/system/mqtt-bridge.service; enabled; vendor preset: disabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-12-17 23:51:47 CET; 889ms ago
    Process: 1980918 ExecStart=/home/root/mqtt-bridge/venv/bin/python3 /home/root/mqtt-bridge/bridge.py (code=exited, status=203/EXEC)
   Main PID: 1980918 (code=exited, status=203/EXEC)
root@WB628598:~/mqtt-bridge#

**Reply by jagheterfredrik** [reply]
Never seen that before, what region are you in?

Can you try running `/usr/bin/python3` on the wallbox?

**Reply by jagheterfredrik** [reply]
Also `apt-cache show python3-core`

**Reply by niouniou49** [reply]
from france

**Reply by niouniou49** [reply]
root@WB628598:/usr/bin# apt-cache show python3-core
N: Unable to locate package python3-core
E: No packages found
root@WB628598:/usr/bin#

**Reply by niouniou49** [reply]
no python3 in usr/bin folder

**Reply by jagheterfredrik** [reply]
Well you have a very unique Pulsar Plus, is it old or new?

Can you run `cat /etc/apt/sources.list`?

**Reply by niouniou49** [reply]
root@WB628598:/usr/bin# cat /etc/apt/sources.list
deb http://pulsar-pm3-repo.wall-box.com/ kirkstone main

**Reply by jagheterfredrik** [reply, marked as answer]
Wow, so it's new. Updated operating system, very nice, Wallbox. I'll have to investigate further, I'll reopen your issue related to this

**Reply by niouniou49** [reply]
can we install manually python3?

**Reply by jagheterfredrik** [reply]
Maybe, can you give me `uname -a` as well?

**Reply by jagheterfredrik** [reply]
Also can you confirm that you bought it recently?

**Reply by jagheterfredrik** [reply]
Can you check inside if the Pi Compute Module is version 4?

**Reply by tronikos** [reply]
Can you try running `python --version` as well?

**Reply by jagheterfredrik** [reply]
Of course 😅 didn’t occur to me

**Comment by niouniou49** [comment]
root@WB628598:~/mqtt-bridge# ./install.sh
Setting up virtual environment
./install.sh: line 24: python3: command not found
Installing Python dependencies
./install.sh: line 27: venv/bin/pip: No such file or directory
Setting up the MQTT bridge systemd service
ln: failed to create symbolic link '/lib/systemd/system/mqtt-bridge.service': File exists
Enable the service to start on boot..
..and launch the service now
Sleeping for 3 seconds before checking the status of the systemd service
● mqtt-bridge.service - MQTT Bridge
     Loaded: loaded (/etc/systemd/system/mqtt-bridge.service; enabled; vendor preset: disabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-12-17 23:51:47 CET; 889ms ago
    Process: 1980918 ExecStart=/home/root/mqtt-bridge/venv/bin/python3 /home/root/mqtt-bridge/bridge.py (code=exited, status=203/EXEC)
   Main PID: 1980918 (code=exited, status=203/EXEC)
root@WB628598:~/mqtt-bridge#

**Reply by niouniou49** [reply]
![Uploading Screenshot 2023-12-17 at 23.54.30.png…]()

**Reply by niouniou49** [reply]
Setting up virtual environment
./install.sh: line 24: python3: command not found
Installing Python dependencies
./install.sh: line 27: venv/bin/pip: No such file or directory

**Reply by niouniou49** [reply]
seems problem on line 24 and 27

**Comment by niouniou49** [comment]
![Uploading Screenshot 2023-12-17 at 22.56.53.png…]()

**Reply by niouniou49** [reply]
i dont know the 172.30.32.1 , all  my internal adress on lan are 10.10.1.XXX.
the wallbox is 10.10.1.120 and the mqtt is 10.10.1.76

**Reply by jagheterfredrik** [reply]
Can you connect to it using e.g. [MQTT Explorer](https://mqtt-explorer.com/)?

**Reply by niouniou49** [reply]
no i can not

**Reply by niouniou49** [reply]
can connect to 10.10.1.76
